manager.go

```diff
@@ -1569,23 +1585,87 @@ func (m *Manager) sanityCheck() error {
 	return nil
 }
 
-// newNetlinkConnection - (TC classifier) TC classifiers are attached by creating a qdisc on the requested
-// interface. A netlink socket is required to create a qdisc. Since this socket can be re-used for multiple classifiers,
-// instantiate the connection at the manager level and cache the netlink socket.
-func (m *Manager) newNetlinkConnection(ifindex int32, netns uint64) (*netlinkCacheValue, error) {
-	var cacheEntry netlinkCacheValue
-	var err error
-	// Open a netlink socket for the requested namespace
-	cacheEntry.rtNetlink, err = tc.Open(&tc.Config{
-		NetNS: int(netns),
-	})
+// NewCachedNetlinkSocket - TC classifiers are attached by creating a qdisc on the requested interface. A netlink socket
+// is required to create a qdisc (or to attach an XDP program to an interface). Since this socket can be re-used for
+// multiple probes, instantiate the connection at the manager level and cache the netlink socket.
+func (m *Manager) NewCachedNetlinkSocket(nsHandle uint64, nsID uint32) (*NetlinkSocket, error) {
```

Comment:
Could you clarify what type of ID the `nsID` value is? Is it the netlink-provided ID, the inode number of the network namespace, something else?

Reply:
Hey 👋🏻 
This is the ID retrieved in `net.ns.inum` (`net` => `struct net`, `ns` => `struct ns_common`, `inum` => `unsigned int`). I'm not exactly sure what you mean by `inode number of the network namespace` because afaik multiple files (thus inodes) could hold a reference to a network namespace ... but maybe there is another inode number I don't know about.
TL;DR I guess this is the netlink-provided ID, if that ID matches the one held by the structures described above 😄 
I added something in the function comment :)

Reply:
That actually is NOT the netlink ID. We call it the inode number in code (https://github.com/DataDog/datadog-agent/blob/e6a153e8812334792fc6489a20fbdb2cae30f05d/pkg/process/util/netns.go#L129), but you can view it from the filesystem with `sudo stat /proc/1/ns/net`:
`File: /proc/1/ns/net -> net:[4026531992]`

Reply:
Docs are limited, but I found this from `ip-netns(8)`
```
ip netns set NAME NETNSID - assign an id to a peer network
       namespace

              This command assigns a id to a peer network namespace.
              This id is valid only in the current network namespace.
              If the keyword "auto" is specified an available nsid will
              be chosen.  This id will be used by the kernel in some
              netlink messages. If no id is assigned when the kernel
              needs it, it will be automatically assigned by the kernel.
              Once it is assigned, it's not possible to change it.
```
The netlink ID only seems to be used when obtaining data over netlink. It is sent as ancillary data. If you want to lookup the mapping between the netlink ID and the inode/other ID, you have to issue another netlink request.